### PR TITLE
feat(generator, factory): updated logic according to the last audit

### PR DESCRIPTION
**Pull request by P-Yevhenii**

### CODE REVIEW

File: contracts/tokenomics/generator/src/contract.rs [hunk 1]
```diff
@@ -891,53 +1269,63 @@ pub fn deposit(
         vec![]
     };
 
-    let reward_msg = build_claim_pools_asset_reward_messages(
+    // let reward_msg = build_claim_pools_asset_reward_messages(
```
Comment (epanchee) [inline contracts/tokenomics/generator/src/contract.rs:1272]: Mark this with // TODO: resolve when bLuna is deployed on Terra2

File: contracts/tokenomics/generator/src/contract.rs [hunk 2]
```diff
@@ -1523,79 +2375,403 @@ pub fn query_simulate_future_reward(
...
+                        total_virtual_supply: lp_balance,
                     };
                     POOL_INFO.save(deps.storage, &key, &pool_info)?;
                 }
+
+                migration::migrate_configs_to_v120(&mut deps, active_pools, &msg)?;
+                migration::migrate_configs_to_v130(deps.storage)?;
+                migration::migrate_configs_to_v200(&mut deps, &msg)?
+            }
+            "1.1.0" => {
```
Comment (epanchee) [inline contracts/tokenomics/generator/src/contract.rs:2614]: I guess migrations for 1.2.0 and 1.3.0 can be removed

File: contracts/tokenomics/generator/tests/integration.rs [hunk 3]
```diff
@@ -588,150 +992,144 @@ fn generator_without_reward_proxies() {
...
+    check_token_balance(&mut app, &lp_eur_usd, &generator_instance, 0);
+    check_token_balance(&mut app, &lp_eur_usd, &user1, 10);
+    check_token_balance(&mut app, &lp_eur_usd, &user2, 10);
 
     check_token_balance(&mut app, &astro_token_instance, &user1, 7_000000);
-    check_token_balance(&mut app, &astro_token_instance, &user2, 6_000000 + 2_000000);
+    check_token_balance(&mut app, &astro_token_instance, &user2, 5_000000);
 }
 
-// TODO: Mirror protocol does not support Terra2 and cosmwasm-1.0 respectively.
```
Comment (epanchee) [inline contracts/tokenomics/generator/tests/integration.rs:733]: Keep these todos

File: packages/astroport/src/lib.rs [hunk 4]
```diff
@@ -20,6 +19,7 @@ pub mod xastro_token;
 
 #[cfg(test)]
 mod mock_querier;
+pub mod pair_stable_bluna;
```
Comment (epanchee) [inline packages/astroport/src/lib.rs:22]: Move this one the previous place

File: scripts/build_release.sh [hunk 5]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
```
Comment (epanchee) [inline scripts/build_release.sh:1]: Give execution rights to this script

**Review by ShadoySV — CHANGES_REQUESTED** [review]

**Review by epanchee — APPROVED** [review]

**Review by ShadoySV — APPROVED** [review]

**Pull request merged**